### ISSUE TITLE
How much memory the A100 you used has?

### ISSUE THREAD

**Issue by Parsifal133** [farewellthree/PPLLaVA#9]
I encountered an out-of-memory issue when trying to perform DPO training on an A800 GPU (80GB memory), even with the batch size set to 1. I noticed that you mentioned training on A100 GPUs. Could you clarify how much memory the A100 you used has?

**Comment by farewellthree** [comment]
We are also using the A100-80G. Perhaps you could try reducing the number of frames to 16 (num_frames parameter) and set the pooling_kernel and pooling_stride to (1, 3, 3)?

**Comment by Parsifal133** [comment]
Thank you for your suggestion, @farewellthree! 

However, I am concerned that adjusting these parameters (num_frames parameter or pooling_kernel size ) might negatively affect the training performance. 

So, i rechecked my environment and was able to train DPO successfully by setting the environment variable `export NCCL_P2P_LEVEL=NVL`.

Due to hardware limitations, I can only use 4 A800 GPUs for training. I set the batch size to 1 (default is 2 in the configuration file) and doubled the gradient accumulation steps.

I would like to know if this training configuration (4 A800 * batch size = 1 * gradient_accumulation_steps=4) can theoretically achieve results consistent with your paper. Or should I reduce the learning rate accordingly (dividing it by 4), considering your paper mentioned using 16 A100 GPUs?

Apologize for not being very familiar with this training framework,  and looking forward to your guidance when you have time.

**Comment by farewellthree** [comment]
Hi, DPO training is not very sensitive to these hyperparameters, but it tends to exhibit considerable fluctuation during training. Typically, the best results are achieved halfway through the training process, so it is necessary to save multiple checkpoints and test them one by one.

**Comment by Parsifal133** [comment]
Thank you for your detailed explanation; it means a lot to me. 

Best wishes for your paper and research!
